_posts/2019-08-13-how-to-deploy-production-grade-vpc-aws.adoc

```diff
@@ -607,18 +608,18 @@ Gruntwork Service Catalog::
 +
 IMPORTANT: You must be a https://gruntwork.io/[Gruntwork subscriber] to access this code.
 +
-Make sure to read <<how_to_use_gruntwork_service_catalog>>.
+Make sure to read link:../gruntwork/how-to-use-gruntwork-service-catalog[How to Use the Gruntwork Service Catalog].
```

Comment:
> I'm guessing that an absolute URL (e.g., /<category>/<title>) would be a bit longer, but probably safer in this case...

Ok,  but the absolute url would be /guides/<cloud>/<category>/<title> then. 

> Not urgent, but it probably makes sense for the URLs to match our breadcrumbs, which do include the cloud... And to have landing pages for each cloud and each category within each cloud. But again, this can be done in future PRs.

I'm not sure if changing the URLs without having landing pages is a good idea. On the other hand it is also not a good idea to launch this with one url structure and then change it. Are we really set on having the cloud as part of the breadcrumbs and the url structure? If this makes a big SEO difference, then it might be worth it to work on these landing pages asap. If not, I'd just drop it from the breadcrumb. I also have another question... Will every guide have a cloud? What about the gruntwork service catalog guide? And the security guide? And other more general clouds?

Reply:
I'm doing some more reflection of the SEO benefits of inserting the cloud in the urls of the guides...

At least for the current guides, it would bring 0 SEO benefit because the keyword would be a duplicate. This is likely to happen very often because we will often want to be somewhat details on the title which has it's own SEO importance.
/guides/networking/how-to-deploy-production-grade-vpc-**aws**
/guides/foundations/how-to-configure-production-grade-**aws**-account-structure
/guides/kubernetes/deploying-a-dockerized-app-on-**gcp**-gke

My conclusion is that it is not worth it for URL SEO benefits. 

The only remaining question is if the landing pages would have its own value for marketing. Which I think they would, but maybe I'm not sure if the guides are the best place for that? Isn't that the plan for the use cases landing pages, be marketing oriented?  

I'm leaning stronger on just removing it from the breadcrumb on design

* faster to ship as it does not require altering url structure or updating breadcrumb (which would be fast) but most importantly does not require making landing pages now (could be a lengthy gatekeeping)
* shorter url with same SEO

Reply:
> Ok, but the absolute url would be/guides/<cloud>/<category>/<title> then. A quick search told me SEO systems prefer that, so it is worth the length.

👍 

> On the other hand it is also not a good idea to launch this with one url structure and then change it later when we add the landing pages, because if we get linked to old urls this would cause another headache to think about. 

True!

> I also have another question... Will every guide have a cloud? What about the gruntwork service catalog guide? And the security guide? And other more general guides?

Good point!

> I'm leaning stronger on just removing it from the breadcrumb on design

Agreed! Nice research 👍

Reply:
Crap... One issue with the absolute URLs: they don't work when GitHub renders the AsciiDoc :(

Reply:
Neither did the other ones. Interdoc ref is just not going to work on github, for the same reasons it would not work on the website.

Reply:
Question, if the repo turns private, why does that matter?

Reply:
Hm, I think you're right; without first-class support for cross-references, it won't work in GitHub. OK, I guess we'll just have to live with that.

It matters if it's private because guides, in the future, will be able to live in any repo. E.g., It makes more sense for the VPC guide to be next to the Ref Arch code (which customers may have access to) and using snippets from it, rather than copy/pasting those snippets. Moreover, we will want to expose the ability for customers to create their own guides in their own repos and for Launchpad to render those too. 

But again, not much we can do here without first class support from AsciiDoc.